sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -45,11 +45,18 @@ private[hive] class SparkExecuteStatementOperation(
     parentSession: HiveSession,
     statement: String,
     confOverlay: JMap[String, String],
-    runInBackground: Boolean = true)
+    runInBackground: Boolean = true,
+    queryTimeout: Long)
   extends ExecuteStatementOperation(parentSession, statement, confOverlay, runInBackground)
   with SparkOperation
   with Logging {
 
+  private val queryTimeoutValue = if (queryTimeout <= 0) {
```

Comment:
@wangyum @AngersZhuuuu Is This behaviour the same with the current Hive one?

Reply:
I am https://github.com/apache/hive/commit/b6218275b00b64aed7efaf470784cc0441464f67#diff-b0085fe84239ae927dfe7464a2174a1c80169bbb60a295e78aeb8df3747e4443R149. 0 seems default meaning no timeout that's matched with the current default's and behaviour here.

Reply:
> @wangyum @AngersZhuuuu Is This behaviour the same with the current Hive one?

When `THRIFTSERVER_QUERY_TIMEOUT ` less then user defined, we respect `THRIFTSERVER_QUERY_TIMEOUT `
https://github.com/apache/hive/blob/940ee46d3a7a9300a47e5beae7c3ca6eb32fd759/service/src/java/org/apache/hive/service/cli/operation/SQLOperation.java#L118-L122

Reply:
Ah, that's what I was looking for. Thanks! @AngersZhuuuu @HyukjinKwon  Updated, could you check the latest fix?